client/src/main/java/org/apache/celeborn/client/read/checkpoint/WorkerPartitionReaderCheckpointMetadata.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.client.read.checkpoint;
+
+import java.util.Set;
+
+/** Checkpoint metadata for a partition reader on the worker side. */
+public class WorkerPartitionReaderCheckpointMetadata implements PartitionReaderCheckpointMetadata {
```

Comment:
This class can be used in worker partition readers, DFS partition reader, and local partition readers. IMO, here is no need to extract an empty interface for it.

Reply:
I initially thought of implementing it only for the WorkerPartitionReader, but I can add no-op impl for other types of readers, and can take up their implementation in a subsequent PR